advisor/src/main/kotlin/advisors/NexusIq.kt

```diff
@@ -78,8 +78,8 @@ class NexusIq(
                 val packageUrl = buildString {
                     append(pkg.purl)
                     val purlType = pkg.id.getPurlType()
-                    if (purlType == Identifier.PurlType.MAVEN) append("?type=jar")
-                    if (purlType == Identifier.PurlType.PYPI) append("?extension=tar.gz")
+                    if (purlType == Identifier.PurlType.MAVEN.toString()) append("?type=jar")
+                    if (purlType == Identifier.PurlType.PYPI.toString()) append("?extension=tar.gz")
```

Comment:
Maybe also turn this into a `when` now?

                    when (pkg.id.getPurlType()) {
                        Identifier.PurlType.MAVEN.toString() -> append("?type=jar")
                        Identifier.PurlType.PYPI.toString() -> append ("?extension=tar.gz")
                    }